### PR TITLE
Disable invocation of policies for missed assets from cloud provider

### PR DESCRIPTION
**Pull request by belosh59**

### CODE REVIEW

**Comment by belosh59** [comment]
So it will search via seqscan till the first found row matching filter and then return. 
**In worst case** if no resources matching cloudProvider - the sequential scan should happen over whole table of assets.

We also query resource_type column for missingAssets analysis, so probably worth to add index.

select count(*) from assets; -- 872

explain analyze SELECT EXISTS(SELECT 1 FROM assets WHERE resource_type like 'AWS%');

```
"Result  (cost=0.17..0.18 rows=1 width=1) (actual time=0.037..0.069 rows=1 loops=1)"
"  InitPlan 1 (returns $0)"
"    ->  Seq Scan on assets  (cost=0.00..148.79 rows=863 width=0) (actual time=0.018..0.026 rows=1 loops=1)"
"          Filter: ((resource_type)::text ~~ 'AWS%'::text)"
"Planning Time: 0.078 ms"
"Execution Time: 0.126 ms"
```

File: magpie-core/src/main/java/io/openraven/magpie/core/cspm/services/PolicyAnalyzerServiceImpl.java [hunk 1]
```diff
@@ -54,74 +52,104 @@ public ScanResults analyze(List<PolicyContext> policies) throws Exception {
...
+      } catch (Exception e) {
+        evalErr.append(e.getMessage());
+      }
+    }
+
+    results.forEach(result -> {
+      Violation violation = new Violation();
+      violation.setPolicyId(policy.getId());
+      violation.setRuleId(rule.getId());
+      violation.setAssetId(result.get("asset_id").toString());
```
Comment (belosh59) [inline magpie-core/src/main/java/io/openraven/magpie/core/cspm/services/PolicyAnalyzerServiceImpl.java:141]: @kickroot I have doubts about line below: we are producing violation information based on policy.description and not rule.description which was violated,
which will then lead to similar violation info for various rules

Comment (belosh59) [reply magpie-core/src/main/java/io/openraven/magpie/core/cspm/services/PolicyAnalyzerServiceImpl.java:141]: But in the end I see that ReportServiceImpl never getting this data for building reports

Comment (kickroot) [reply magpie-core/src/main/java/io/openraven/magpie/core/cspm/services/PolicyAnalyzerServiceImpl.java:141]: Don't worry about this for the moment, I'll verify the output columns separately.

**Review by kickroot — APPROVED** [review]

**Pull request merged**